src/common/options/rgw.yaml.in

```diff
@@ -3078,6 +3078,16 @@ options:
   services:
   - rgw
   with_legacy: true
+- name: rgw_crypt_sse_s3_vault_key_rotation
+  type: str
+  level: advanced
+  desc: The period at which this key should be rotated automatically
+  long_desc: Setting this to "0" (the default) will disable automatic key rotation.
+    This value cannot be shorter than one hour. Uses duration format strings.
```

Comment:
this either needs to describe the supported duration format string, or just define the configurable as an integer in seconds and add the "s" when sending to vault

Reply:
@cbodley Do you think mentioning this doc would be a good idea? https://developer.hashicorp.com/vault/docs/concepts/duration-format

Reply:
i think it's better to make the config option an integer value in seconds instead. that way rgw can always send a valid format string to vault

if we leave that up to the admin, they could accidentally mess up the format string, and Ceph wouldn't catch problems until runtime when sse-s3 uploads start failing due to errors from vault. at that point, it could be very hard to tell that the failures are caused by this format string

Reply:
It's fine for me - but just one point would be because mostly the value for this config is in months or years then having a long int in seconds might not be human-readable for config files - We might consider letting the admins trust their orchestrators and pipelines to put a correct config (which this config is not gonna change a lot I guess =) ).
@cbodley What do you think? Are you still in favor of seconds?

Reply:
given that https://developer.hashicorp.com/vault/api-docs/secret/transit#auto_rotate_period says:
> This value cannot be shorter than one hour

it would be reasonable to use hours instead of seconds. does that help?

Reply:
Then if someone wanted to mix hours with minutes or seconds? :D

Reply:
I mean even with seconds with the quoted text - if admin set for example 2 seconds in the config again it gets failure. So we should at some point let admin trust/examine what he is doing? 🤔

Reply:
> I mean even with seconds with the quoted text - if admin set for example 2 seconds in the config again it gets failure.

sticking to hours sounds the least error-prone then. would ceph admins really need to control key rotation down to the second?

i could see that being useful when writing regression tests under qa/suites/rgw/crypt, but we can't wait an entire hour for the first rotation to happen

Reply:
@cbodley Done. Please take a look at the implementation.
Basically, I was trying to avoid doing something like this (https://github.com/ceph/ceph/pull/53278/files#diff-4edf36573eb025afac5c9ef719d00c1dd0e5504d8116341f2beea0ec3c59289cR627) on every call of the function. 😅